### DISCUSSION TITLE
An attempt to lighten the burden of a cross-platform GUI: Why not a Text User Interface?

### DISCUSSION THREAD

**Discussion by rflcrz** [espanso/espanso#1203] [Feature requests and Ideas]
Espanso brings together features that make it unique: it's open, cross-platform, and it's text-based! This makes it extensible in many ways, thus powerful!

Considering that "creating a cross-platform GUI is not a trivial thing" (#255), why not create a TUI (text user interface)? This would make Espanso one of a kind and (hopefully) lessen the burden of a cross-platform UI!

I am not proposing to abandon the idea of a cross-platform GUI; [modulo](https://github.com/federico-terzi/modulo) is definitely needed to provide user interaction during expansion! Also, modulo will make Espanso more than "a tool for the tech-savvy" (#255), making it more friendly to the rest of us (I'm not a developer).

But instead, I’m proposing to add a TUI to espanso-core (or make it a separated project, depending on the complexity).

The text interface would be available to those comfortable with a terminal (even on a remote session) and a window wrapper could be made for (convenience of) those who are not comfortable opening a terminal.

Personally, I would tend to use a TUI over a GUI, given the benefits of a keyboard-centric nature of a text interface — which in the long run would make me faster, thus productive.

In this sense, depending on the usability achieved with a text interface, a TUI could be considered as a replacement to a GUI (for the purpose of configuration).

For the reasons presented above I understood that a TUI is in harmony with the spirit of Espanso — cross-platform productivity and expansibility; which led me to this text interface library for Rust, called [Cursive](https://github.com/Gyscos/Cursive).

And, as a physician who is not able to contribute with code, I couldn't help but take the opportunity to thank the developers for their efforts in providing this nice piece of software!

**Comment by federico-terzi** [comment]
Hey @rflcrz,

First of all, thank you for the detailed proposal! I agree with you on the fact that a TUI would pretty cool, and also a good companion for espanso.
That said, although espanso was designed as a tool for the tech-savvy (and in a way, it will always be), I really want to make it more accessible to other non-technical people, and for that, a GUI is necessary. Honestly, even though I consider myself a power user, there are times in which I would just prefer a GUI when working with espanso, so that for simple matches I don't have to deal with YAML :)
I'm not saying that a TUI will never come, but it definitely has a lower priority than other features, as tech-savvy people (the ones that would most benefit from a TUI) can edit the config files directly (although that is inconvenient at times). 

Of course, if someone from the community is willing to create a TUI for espanso, that could be a really nice side project! :)

**Comment by reesericci** [comment]
I would love a TUI !
